File: backend-go/internal/modules/system/entity/sys_role.go
package entity

import (
	"drip-admin/backend-go/internal/common"
	"time"
)

type SysRole struct {
	ID        common.Int64String `gorm:"column:id;primaryKey" json:"id"`
	RoleName  string             `gorm:"column:role_name" json:"roleName"`
	RoleCode  string             `gorm:"column:role_code" json:"roleCode"`
	Builtin   int                `gorm:"column:builtin" json:"builtin"`
	Status    int                `gorm:"column:status" json:"status"`
	Remark    *string            `gorm:"column:remark" json:"remark"`
	Deleted   int                `gorm:"column:deleted" json:"deleted"`
	CreatedAt *time.Time         `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt *time.Time         `gorm:"column:updated_at" json:"updatedAt"`
}

func (SysRole) TableName() string { return "sys_role" }

File: backend-python/app/modules/system/entity/sys_role.py
from sqlalchemy import BigInteger, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.modules.system.entity.base import Base, TimestampMixin


class SysRole(TimestampMixin, Base):
    __tablename__ = "sys_role"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    role_name: Mapped[str] = mapped_column(String(64))
    role_code: Mapped[str] = mapped_column(String(64))
    builtin: Mapped[int] = mapped_column(Integer)
    status: Mapped[int] = mapped_column(Integer)
    remark: Mapped[str | None] = mapped_column(String(500), nullable=True)
    deleted: Mapped[int] = mapped_column(Integer)

